weave/scorers/llm_scorer.py

```diff
@@ -77,86 +78,33 @@ def validate_client(cls, v: _LLM_CLIENTS) -> _LLM_CLIENTS:
         return instructor_client(v)
 
 
-class HuggingFacePipelineScorer(Scorer):
-    """
-    Base class for using Hugging Face pipelines for moderation scoring.
-
-    This class simplifies the use of Hugging Face pipelines by handling the initialization and providing a common interface for scoring.
-
-    Args:
-        task (str): The pipeline task type (e.g., `"text-classification"`).
-        model_name_or_path (str): The name or path of the model to use.
-        device (str): The device to use for inference. Defaults to `"cpu"`.
-        pipeline_kwargs (dict[str, Any]): Additional keyword arguments for the pipeline. Defaults to `{}`.
-
-    Returns:
-        list[dict[str, Any]]: The pipeline's output after processing the input text.
-
-    Example:
-        >>> from weave.scorers.moderation_scorer import PipelineScorer
-        >>> scorer = PipelineScorer(
-        ...     task="text-classification",
-        ...     model_name_or_path="distilbert-base-uncased-finetuned-sst-2-english"
-        ... )
-        >>> output = scorer.pipe("This is a great movie!")
-        >>> print(output)
-        [{'label': 'POSITIVE', 'score': 0.9998}]
-    """
-
-    task: str = Field(
-        description="The task to use for the pipeline, for example 'text-classification'"
-    )
-    model_name_or_path: str = Field(default="", description="The path to the model")
-    device: str = Field(default="auto", description="The device to use for the model")
-    pipeline_kwargs: dict[str, Any] = Field(default_factory=dict)
-    pipeline: Optional[Any] = None
-
-    def model_post_init(self, __context: Any) -> None:
-        self.device = set_device(self.device)
-        try:
-            import transformers
-        except ImportError:
-            print(
-                "The `transformers` package is required to use PipelineScorer, please run `pip install transformers`"
-            )
-        if self.pipeline is None:
-            self.load_pipeline()
-
-    def load_pipeline(self) -> None:
-        raise NotImplementedError(
-            "Subclasses must implement the `load_pipeline` method."
-        )
-
-    @weave.op
-    def score(self, *, output: Any, **kwargs: Any) -> Any:
-        raise NotImplementedError
-
-
 class HuggingFaceScorer(Scorer):
     """Score model outputs using a Hugging Face model."""
 
-    model_name_or_path: str = Field(default="", description="The path to the model")
+    model_artifact_or_path: Union[str, Path] = Field(
+        default="", description="The path to the model"
+    )
     device: str = Field(default="auto", description="The device to use for the model")
-    model: Any = None
-    tokenizer: Any = None
+    _model: Any = PrivateAttr(default=None)
```

Comment:
Since this is a `PrivateAttr`, it won't get serialized so you can be more specific with the type, like `PreTrainedModel`

Reply:
that is the intention, right. These properties are calculated post init?